oshmem/mca/sshmem/mmap/sshmem_mmap_module.c

```diff
@@ -184,17 +184,48 @@ segment_create(map_segment_t *ds_buf,
     /* init the contents of map_segment_t */
     shmem_ds_reset(ds_buf);
 
-    addr = mmap((void *)mca_sshmem_base_start_address,
+    if (mca_sshmem_mmap_component.is_anonymous) {
+        addr = mmap((void *)(mca_sshmem_mmap_component.is_start_addr_fixed ? mca_sshmem_base_start_address : NULL),
                 size,
                 PROT_READ | PROT_WRITE,
                 MAP_PRIVATE |
-#if defined(MAP_ANONYMOUS)
```

Comment:
is it enough?
https://github.com/miked-mellanox/ompi-release/blob/topic/v1_10_oshmem_update/oshmem/mca/sshmem/mmap/sshmem_mmap_module.c#L52